File: tests/test_wip.cpp
#include <pybind11/pybind11.h>

#include "pybind11_tests.h"

namespace pybind11_tests {
namespace wip {

struct Vanilla {};

} // namespace wip
} // namespace pybind11_tests

TEST_SUBMODULE(wip, m) {
    using namespace pybind11_tests::wip;

    py::class_<Vanilla>(m, "Vanilla", py::metaclass((PyObject *) &PyType_Type)).def(py::init<>());
}

File: tests/test_wip.py
import abc

from pybind11_tests import wip as m


class VanillaABC(m.Vanilla, metaclass=abc.ABCMeta):
    pass


def test_vanilla():
    assert m.Vanilla().__class__.__name__ == "Vanilla"
    assert type(m.Vanilla).__name__ == "type"
